### DISCUSSION TITLE
New project and Csla 6.2

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3238] [Questions]
We have a (very) small project coming up :)
and I am thinking of using Csla 6.2. Did we upgrade project tracker? just wanted to look at the latest business library code.

Kind Regards

**Comment by kcabral817** [comment]
There are a couple of other projects that might be useful.  The two that I have used are BlazorServer and the dataportal example.  I just converted an enterprise wpf to blazor using Syncfusion controls.   The biggest thing hurdle was to ensure the ApplicatonContext was not null when instantiating an object.  Other than that, the framework is still the same.  If you want to have communicate directly, let me know and I can show you what I did if you'd like.

**Reply by Chicagoan2016** [reply]
Thank you Sir, We are going to decide the UI, which believe it or not,  might end up Winforms lol. I appreciate your help and I will contact you.
Kind Regards.

**Reply by rockfordlhotka** [reply]
The Windows Forms sample is current. Be warned that the sample uses DI, and so is a little different from the old style WinForms apps.

You can still use the old style - though you need to expose the service provider (and/or `ApplicationContext`) as a global property so the whole app has access to it.

If you are writing a new app, I'd probably give serious consideration to the new style using DI, because that'll make so many things easier.

**Reply by TanguyIngels** [reply]
@kcabral817 I'm facing the same issue like you described working with syncfusion controls.  When editing a record, the ApplicationContext == null.  So behind the scenes they're doing some copy of data.  How did you solve that ApplicationContext is still known when using inline editing in the syncfusion controls? Thank you for sharing your solutions.

**Reply by rockfordlhotka** [reply]
@TanguyIngels just thinking out loud, I wonder if we could mitigate this somewhat.

If you _know for certain_ that you are running in a single-scope environment (from a DI perspective), such as a WinForms app - where the app is running in one DI scope for everything (except logical server-side data portal calls), _maybe_ CSLA could have some way for you to provide a global callback function to get at the `ApplicationContext` instance for the app.

Lots of caveats obviously!

What I'm thinking specifically, is that if a domain object needs `ApplicationContext`, and the `ApplicationContext` is `null`, then it would call this global function (somehow) to get the current `ApplicationContext` - and you'd provide that value from the global `App.ApplicationContext` that you are maintaining.

Of course this might reveal other issues - I'm just playing with ideas here 😇

**Reply by TanguyIngels** [reply]
@rockfordlhotka .  Your last sentence cauhgt my attention "reveal other issues".  So I've been debugging a bit in my Blazor project, but the third party tool "SyncFusion" is creating a new instance (not via the Csla-way), with result that the ApplicationContext isn't known and also if it's concerning a child, the parent is unknown.  So the only solution will be that the third party offers some overridable methods so I can handle myself what must be done when a CRUD operation is required.  

So the solution for me will to not use the inline editing feature of their Grid control, and do some custom stuff.  IIf I do wish to use inline editing, I will submit a feature to Syncfusion.  But inline editing can give quick results in some situations, but as far as I am concerned from some customisation, you should still move away from this.

Thanks for your reply, has given me pause to think which way I want to proceed.

**Comment by rockfordlhotka** [comment]
Sadly I haven't updated ProjectTracker yet, but @kcabral817 has good suggestions. Pretty much all the more focused sample projects are updated (or close) for 6.2.

**Comment by kcabral817** [comment]
I didn't know that you were using the grid.  That does not work and you won't be able to intercept it.  I know, I tried for several days to get that to run.

Have a nice day.

Kevin

https://rb.gy/syiuf


________________________________
